apps/confidential/confidential_test.go

```diff
@@ -107,6 +107,25 @@ func fakeClient(tk accesstokens.TokenResponse, credential Credential, options ..
 	return client, nil
 }
 
+func fakeClientNoInstanceDiscovery(tk accesstokens.TokenResponse, credential Credential, options ...Option) (Client, error) {
+	options = append(options, WithAuthority("https://fake_authority/fake"))
+	options = append(options, WithInstanceDiscovery(false))
+	client, err := New("fake_client_id", credential, options...)
+	if err != nil {
+		return Client{}, err
+	}
+	client.base.Token.AccessTokens = &fake.AccessTokens{
+		AccessToken: tk,
+	}
+	client.base.Token.Authority = &fake.Authority{}
```

Comment:
`Authority` is responsible for instance discovery, so this fake is replacing code we want to test. I recommend instead using the `mock` package to test an unmodified client end-to-end, through its public API, as in [TestWithTenantID](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/4abfea36f99928ad52523e017b804616993470f5/apps/confidential/confidential_test.go#L713).